elliott

```diff
@@ -1038,7 +1038,143 @@ def find_cve_trackers(runtime, cve, status):
     for b in bug_list:
         click.echo("{}\t{:15s}\t{}".format(b.bug_id, b.status, b.summary))
 
+#
+# Verify images in a payload match the corresponding advisory
+# verify-payload
+#
+@cli.command("verify-payload", short_help="List builds attached to ADVISORY")
+@click.option("--payload", metavar="PULLSPEC", required=True,
+              help="The pullspec to the payload to verify")
+@click.argument('advisory', type=int)
+@click.pass_context
+def verify_payload(ctx, payload, advisory):
+    """Cross-check that the builds present in PAYLOAD match the builds
+attached to ADVISORY.
+
+Results are summarily printed at the end of the run. They are also
+written out to summary_results.json.
 
+     Verify builds in the given payload match the builds attached to
+     advisory 41567
+
+ \b
+    $ elliott verify-payload --payload registry.svc.ci.openshift.org/ocp/release:4.1.0-0.nightly-2019-05-22-190823 41567
+"""
+    try:
+        green_prefix("Fetching advisory builds: ")
+        click.echo("Advisory - {}".format(advisory))
+        builds = elliottlib.errata.get_builds(advisory)
+    except GSSError:
+        exit_unauthenticated()
+    except elliottlib.exceptions.ErrataToolError as ex:
+        raise ElliottFatalError(getattr(ex, 'message', repr(ex)))
+
+    all_advisory_nvrs = {}
+    # Results come back with top level keys which are brew tags
+    green_prefix("Looping over tags: ")
+    click.echo("{} tags to check".format(len(builds.keys())))
+    for tag in builds.keys():
+        # Each top level has a key 'builds' which is a list of dicts
+        green_prefix("Looping over builds in tag: ")
+        click.echo("{} with {} builds".format(tag, len(builds[tag]['builds'])))
+        for build in builds[tag]['builds']:
+            # Each dict has a top level key which might be the actual
+            # 'nvr' but I don't have enough data to know for sure
+            # yet. Also I don't know when there might be more than one
+            # key in the build dict. We'll loop over it to be sure.
+            for name in build.keys():
+                split_name = name.split('-v4')[0]
```

Comment:
I do not like assuming that `-v4` is the version.
There is a somewhat canonical way to split out an nvr:
```suggestion
                name, version, release = name.rsplit('-', 2)
```

Reply:
Dang you good.